### ISSUE TITLE
error MSB6006: "csc.exe" exited with code 8

### ISSUE THREAD

**Issue by Saibamen** [dotnet/roslyn#22663]
**Version Used**: .NET Command Line Tools (2.0.2)

I can't build my project in Linux.
Build log: https://circleci.com/gh/Saibamen/BriskChat/66
Project: https://github.com/Saibamen/BriskChat

**Comment by Saibamen** [comment]
This same with microsoft.net.compilers 2.4.0

**Comment by Leon99** [comment]
The same here, Microsoft.Net.Compilers 2.4.0. All good when the package is removed.

**Comment by Saibamen** [comment]
Build log when I deleted Microsoft.Net.Compilers:
https://circleci.com/gh/Saibamen/BriskChat/81

Build process is ok, but I can't test

**Comment by jcouv** [comment]
@Saibamen 
From your description, my natural inclination is to assume there is a problem with your project, not with the compiler. 
Would you be able to narrow down the repro somewhat, to support the idea that there is a problem with the compiler or compiler package?

**Comment by Saibamen** [comment]
@jcouv 
Today I've recreated solution from scratch - created new projects with newest Visual Studio 15.4.
Commit: https://github.com/Saibamen/BriskChat/commit/5cfb5b1ddbd18e99505908fb685a7689a4444aab
Build log: https://circleci.com/gh/Saibamen/BriskChat/93

With Microsoft.Net.Compilers 2.4.0:
Commit: https://github.com/Saibamen/BriskChat/commit/03eb1297c73b1c88decc7ff8e06629110035aa7e
Build log: https://circleci.com/gh/Saibamen/BriskChat/94

**Comment by jcouv** [comment]
@Saibamen Thanks for the comparative results.

I checked out both commits on Windows, loaded the solution and rebuilt solution from Visual Studio. The build succeeded in both cases. Let me know if that's not the case for you.

I'm trying on Linux (Ubuntu 16 in Docker), as it seems that's what your logs indicate. But commit 5cfb5b1 doesn't pass `dotnet build`.

Here's what I ran:
```
   27  apt-get install dotnet-sdk-2.0.2
   28  dir
   29  ls
   30  git clone https://github.com/Saibamen/BriskChat.git
   31  cd BriskChat/
   32  git checkout 5cfb5b1
   33  dir
   34  dotnet build
```

Produces
`/BriskChat/TrollChat.BusinessLogic/TrollChat.BusinessLogic.csproj.metaproj : error MSB4025: The project file could not be loaded. Could not find file '/BriskChat/TrollChat.BusinessLogic/TrollChat.BusinessLogic.csproj.metaproj'`

Assuming that `dotnet build` produces the problem for you, could you run `dotnet build /bl` instead and share the resulting `msbuild.binlog` file?

**Comment by Saibamen** [comment]
@jcouv 
Thanks for investigating my problem.

I think you must run `dotnet restore` before build ;)

I'm able to build project in Windows 10 with latest Visual Studio, JetBrains Rider. AppVeyor runs it correctly too: https://ci.appveyor.com/project/Saibamen/briskchat/build/1.0.128

Only `dotnet build` (build full solution, builds separate projects: DataAccess, BusinessLogic and BusinessLogic.Tests works fine, but Web project not...) and dotnet test (any options) fails in Travis and CircleCI.

Travis builds: https://travis-ci.org/Saibamen/BriskChat/builds

Here's log for you from `dotnet build -v diag`: https://circleci.com/api/v1.1/project/github/Saibamen/BriskChat/106/output/104/0?file=true

I cannot migrate yet to .NET Core 2.0, because I'm waiting for 2.1 - my project is using SignalR but I think this is not problem with .NET Core but with dotnet builds tools

**Comment by Leon99** [comment]
My project is Core 2.0

**Comment by WilliamWsyHK** [comment]
I am experiencing this issue on macOS 10.13.2 (High Sierra)

when run `dotnet run` or `dotnet build` on command line (I am using default bash Terminal.app), it failed with 

> error MSB6006: "csc.exe" exited with code 8

if the project has `Microsoft.Net.Compilers` as dependency, but the same project works perfectly when without this same dependency, or by using Visual Studio for Mac with the dependency. I suppose this is somehow working due to Mono, since debugging in VS for Mac will start Mono (known by AV is complaining if Mono could access my project location).

**Comment by Ghostbird** [comment]
I have a similar problem. I can't share much info on the proprietary sources, but we have no problem building:
- Windows 10 using Visual Studio Code
- Windows 10 using command line `dotnet build` (arguably identical to using VSCode)
- Windows 10 using Visual Studio
- Debian 9 (Stretch) using Visual Studio Code
- Debian 9 using command line `dotnet build`

We DO have this problem when running on:
- Docker image microsoft/dotnet:2-sdk (or variants thereof) both locally on Debian 9 and on Circle CI

Does anyone have a workaround?

**Comment by jcouv** [comment]
After investigating another CLI issue, I think there are two problems here:
1. CLI doesn't print the diagnostics from the compiler (I believe that issue is tracked by https://github.com/dotnet/roslyn/issues/24736 and a fix is in the works)
2. the compilation fails for some reason

Here are some steps to see the reason for the compilation failing:
1. run the problematic `build` command, but add the option for binary logging `/bl` (for example, `dotnet build /bl`)
2. assuming this reproduced the issue, open the `msbuild.binlog` file with the [binary log viewer](https://github.com/KirillOsenkov/MSBuildStructuredLog).
3. the log viewer should show the failed `Csc` task (as shown below)
4. you can view and copy the command-line arguments to a text file `repro.rsp`, removing the first two chunks (the first one for "dotnet.exe" and the second one for "csc.dll")
5. re-use those two chunks to run `dotnet.exe csc.dll @repro.rsp`
6. this should give you the compiler diagnostics

![image](https://user-images.githubusercontent.com/12466233/36326802-42886daa-1311-11e8-87bd-14c52067fcb9.png)

While I encourage you to try this to unblock yourself, I will try and repro the specific issue described in OP again.

**Comment by SwadX** [comment]
I am facing the same error in amazon linux . 
<sudo dotnet build --no-restore --verbosity d --no-dependencies
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 11/27/18 11:42:22 AM.
     1>Project "/var/lib/jenkins/workspace/multibranch_pipeline_master-VFQZN2R7JTITZDJOVBS6UDCECP2XJL7TD55373F64OTKZXE63T4Q/source/WebServer/UP/DotNetAPI/DotNetAPI.csproj" on node 1 (Build target(s)).
     1>Building with tools version "15.0".
       Project file contains ToolsVersion="". This toolset may be unknown or missing, in which case you may be able to resolve this by installing the appropriate version of MSBuild, or the build may have been forced to a particular ToolsVersion for policy reasons. Treating the project as if it had ToolsVersion="15.0". For more information, please see http://go.microsoft.com/fwlink/?LinkId=293424.
     1>Target "Build" in project "/var/lib/jenkins/workspace/multibranch_pipeline_master-VFQZN2R7JTITZDJOVBS6UDCECP2XJL7TD55373F64OTKZXE63T4Q/source/WebServer/UP/DotNetAPI/DotNetAPI.csproj" (entry point):
       Using "Csc" task from assembly "/opt/dotnet/sdk/2.0.0/Roslyn/Microsoft.Build.Tasks.CodeAnalysis.dll".
       Task "Csc"
         /opt/dotnet/sdk/2.0.0/Roslyn/csc.exe /out:/DotNetAPI.dll /target:library ReliabilityMain.cs
     1>/var/lib/jenkins/workspace/multibranch_pipeline_master-VFQZN2R7JTITZDJOVBS6UDCECP2XJL7TD55373F64OTKZXE63T4Q/source/WebServer/UP/DotNetAPI/DotNetAPI.csproj(10,5): error MSB6006: "csc.exe" exited with code 8.
       Done executing task "Csc" -- FAILED.
     1>Done building target "Build" in project "DotNetAPI.csproj" -- FAILED.>

How to remove the compiler package  Microsoft.Net.Compilers ?

My .csproj file has the below 
<
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
   <TargetFramework>netcoreapp2.0</TargetFramework>
 </PropertyGroup>
  <ItemGroup>  
    <Compile Include="Main.cs" />
    <PackageReference Include="Microsoft.Net.Compilers" Version="2.4.0" />
  </ItemGroup>  
  <Target Name="Build">  
    <Csc Sources="@(Compile)"
    TargetType="library"
     OutputAssembly="$(builtdir)\$(MSBuildProjectName).dll"
    />    
  </Target>  
</Project>  
>

**Comment by SwadX** [comment]
My .csproj file has the below 

> <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
>   <PropertyGroup>
>    <TargetFramework>netcoreapp2.0</TargetFramework>
>  </PropertyGroup>
>   <ItemGroup>  
>     <Compile Include="Main.cs" />
>     <PackageReference Include="Microsoft.Net.Compilers" Version="2.4.0" />
>   </ItemGroup>  
>   <Target Name="Build">  
>     <Csc Sources="@(Compile)"
>     TargetType="library"
>      OutputAssembly="$(builtdir)\$(MSBuildProjectName).dll"
>     />    
>   </Target>  
> </Project>

**Comment by jcouv** [comment]
@Manzanit0 @SwadX If you are still seeing such errors without diagnostics, please file a new issue with details for your repro. In particular, was version of dotnet CLI and compiler.

I've also shared some instructions [earlier](https://github.com/dotnet/roslyn/issues/22663#issuecomment-366344832) to let you dig up the missing diagnostics, if that helps.
